xclim/indices/generic.py

```diff
@@ -89,8 +89,8 @@ def select_resample_op(
     r = da.resample(time=freq)
     if isinstance(op, str):
         return getattr(r, op)(dim="time", keep_attrs=True)
-
-    return r.map(op)
+    with xr.set_options(keep_attrs=True):
+        return r.map(op)
```

Comment:
Oh I see! Now that this function is used alone in some indices, I think it would make sense for it to manage the output units as well. There are many indices in this PR that could be removed if `select_resample_op` could manage units.

@Zeitsperre  As for preserving the other attributes, wouldn't this go against the usual behaviour of xarray ? It makes sense to me that indices drop the attributes, as this is what most xarray operations do. Users can easily change that with the decorator in their own code.

@coxipi Indeed, `doymax` is a good idea. It would set the xclim-specific `is_dayofyear=np.int32(1)`.

Reply:
Thinking of it, I believe `select_resample_op` should explicitly _remove_ the `units` if "op" is unknown. Or `to_agg_units` should do that, but the goal would be to avoid as much as possible the inadvertent preservation of "units". An output without a units attr will make `declare_units` fail and hopefully, this would happen as early as possible, when developing the indicator.

Reply:
@aulemahal Good point. I figured the use of `keep_attrs` above meant that we should be preserving attributes within these operations. Perhaps not neccessary.

Reply:
Maybe `doymax` is too specific for the current `to_agg_units`? I don't think it fits with the current paradigm:
```python 
    m, freq_u_raw = infer_sampling_units(orig[dim])
    out = out * m
```

Reply:
Indeed it doesn't, my suggestion is to extend it so it does!

Genre:
```python
def to_agg_units(...):
    if op in ['doy', 'doymax', 'doymin']:
        # put doy stuff
    elif op in ['min', 'max', 'mean']:
        # put same units
    else:
        # to_agg_units as it exists currently
```

Reply:
En fait, I could do this extension in my PR. This PR will work without it.

Reply:
I wanted to make sure that `select_resample_op` gives the correct units, so it can be used in indices, so I think I need this, no?